components/mediation/tasks/org.wso2.micro.integrator.ntask.core/src/main/java/org/wso2/micro/integrator/ntask/coordination/task/ClusterNodeDetails.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.micro.integrator.ntask.coordination.task;
+
+import org.wso2.micro.integrator.coordination.ClusterCoordinator;
+
+import java.util.List;
+
+/**
+ * The class which provides the cluster node details. No additional controlling methods should be added to this un
+ * necessarily as this is exposed to task location resolver. Task location will be resolved only in leader nodes.
+ *
+ * @see org.wso2.micro.integrator.ntask.coordination.task.resolver.TaskLocationResolver
+ */
+public class ClusterNodeDetails {
```

Comment:
I feel like this name is a bit confusing. From the name along it seems like it doesn't belong to the ntask module and it wrapps a coordinator. Can't we have a better name like 'ClusterCoordinator' may be? 'NodeDetails' seems like just a class that has node details. But this would basically be our delegator for the org.wso2.micro.integrator.coordination.ClusterCoordinator.

Reply:
ClusterCoordinator is fine but we have the same class in coordination as well . What if we use ClusterConnector ? or any other better suggestion